spec/support/warnings.rb

```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new("2.7.2")
+  Warning[:deprecated] = true
+end
```

Comment:
Since Ruby 2.7.2, deprecation warnings aren't shown automatically.

- [Feature #17000](https://bugs.ruby-lang.org/issues/17000) 2.7.2 turns off deprecation warnings by default
- [Feature #16345](https://bugs.ruby-lang.org/issues/16345) Don’t emit deprecation warnings by default.

Enable them during the test suite run so we can identify and resolve API deprecations proactively.